### DISCUSSION TITLE
Feedback from my first live class

### DISCUSSION THREAD

**Discussion by dmumford** [netless-io/flat#898] [General] [answered]
The student initially had issues loading the materials in the room. It took a long time to load the files. They also told me that they were unable to enter the room on their tablet computer, but it worked on their pc.

It took me about 7 attempts connecting to the room when the other person was already in it before it let me in. I kept receiving the following error:

https://ibb.co/fx1Cbd8

I was using a pdf file in the room. There were some discrepancies between myself and the student. For example, when I changed the pdf page, it didn't always change the page for the student. This made it difficult to know which page the student was viewing.

At one point in the class, the drawing tool was severely mis-calibrated. I had to refresh the page a couple of times to get it working again. 

On one occasion,  I played a video in the class but the student did not see it. It worked after a few attempts.

Overall, I really like the tools and function of Flat. However, I think these issues need to be addressed before I get fully behind it. 

By the way, thanks for all of the developers hard work! You've all done great so far.

**Comment by BlackHole1** [comment]
Is it convenient to tell your region (just the name of the city is fine)
We currently feel that it should be caused by network communication, but need your region to facilitate our simulation environment testing

> I was using a pdf file in the room. There were some discrepancies between myself and the student. For example, when I changed the pdf page, it didn't always change the page for the student. This made it difficult to know which page the student was viewing.

> On one occasion, I played a video in the class but the student did not see it. It worked after a few attempts.

@Moskize91 @leavesster Does this happen when the network communication is not very stable?


> At one point in the class, the drawing tool was severely mis-calibrated. I had to refresh the page a couple of times to get it working again.

@dmumford Can you elaborate on this situation?


> They also told me that they were unable to enter the room on their tablet compute

@dmumford is Apple iPad?

**Reply by dmumford** [reply]
I'm in the UK and my student is in China. I've used other platforms which are also built on Agora and we've had no network issues before. 

I didn't notice an unstable network when playing the media files. 

What I mean by mis-calibrated is when I draw using the pen tool, the drawing isn't at the location of my mouse. It's not even close (sometimes off screen). This issue was fixed by refreshing. Maybe it occurs on the web version of Flat when the browser window is resized?

**Reply by BlackHole1** [reply]
Can you provide the UUID of the room? We do not enter the room, it is only used to troubleshoot the problem. If privacy is involved, you can send the room uuid to `flat@agora.io` email.

**Reply by BlackHole1** [reply]
room uuid => room number / room id

**Reply by BlackHole1** [reply]
For your privacy, we have deleted your response in case others have access to it. We have logged it on our internal system. We will run a check tomorrow and will keep you informed if we have any updates.

Thank you for your feedback, your feedback will make us better

**Comment by leavesster** [comment]
Thank you for you feedback.

## need reconnecting to the room multiple times.

>It took me about 7 attempts connecting to the room when the other person was already in it before it let me in. I kept receiving the following error:

Cloud you try to use `dig` command to dig `api.netless.link`.
It will response like this:

```shell
➜  netless dig api.netless.link

; <<>> DiG 9.10.6 <<>> api.netless.link
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 48036
;; flags: qr; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;api.netless.link.		IN	A

;; ANSWER SECTION:
// we need this answer ip 
api.netless.link.	30	IN	A	198.18.2.87

;; Query time: 0 msec
;; SERVER: 198.18.0.2#53(198.18.0.2)
;; WHEN: Thu Aug 26 23:08:12 CST 2021
;; MSG SIZE  rcvd: 66
```

You can alos send us this response to our email `flat@agora.io` which we mention above. or you can just paste response here

## pdf and video sync promble

When you create a room you need choose a region. We need to know which region do you choose.(If you already send us a mail with you problem room. This is not needed any more, we can find it by this room uuid).

**Reply by dmumford** [reply]
; <<>> DiG 9.10.6 <<>> api.netless.link +trace
;; global options: +cmd
.			65581	IN	NS	h.root-servers.net.
.			65581	IN	NS	i.root-servers.net.
.			65581	IN	NS	j.root-servers.net.
.			65581	IN	NS	k.root-servers.net.
.			65581	IN	NS	l.root-servers.net.
.			65581	IN	NS	m.root-servers.net.
.			65581	IN	NS	a.root-servers.net.
.			65581	IN	NS	b.root-servers.net.
.			65581	IN	NS	c.root-servers.net.
.			65581	IN	NS	d.root-servers.net.
.			65581	IN	NS	e.root-servers.net.
.			65581	IN	NS	f.root-servers.net.
.			65581	IN	NS	g.root-servers.net.
.			65581	IN	RRSIG	NS 8 0 518400 20210908170000 20210826160000 26838 . J+s2B48gOmu2v3fBtYVA1v8Eob6X3JzJjWgQDs66ypEkQVatqAhZ6lGg Gyigb50yrt2cWePod3eIgrAUXHRhjCanpCmKeIhwz6HANcITtyNEzz07 wxEszSpeF9JoWFWs5z5StX8bNjDVInOMB2twkdoN1bRfC5v2rCe7fKre y/Qq40cBf5+8CH+hOGIYQ9KcF+a5ClYWylUtCXsJyuZuf1i0bGfqeNgs XHeQyFbBkC7VpN/xMjRtWGLWzZtRRPmzZ7puQvX42tPsewuVsKguS1Db xIctAcW9pwXGPYn9M+SEI+/UF/cZbhJdNw1zc/Fuo/4ljF1TukNkUXiK 3Esyrw==
;; Received 525 bytes from 192.168.1.1#53(192.168.1.1) in 35 ms

link.			172800	IN	NS	ns1.uniregistry.net.
link.			172800	IN	NS	ns2.uniregistry.info.
link.			172800	IN	NS	ns3.uniregistry.net.
link.			172800	IN	NS	ns4.uniregistry.info.
link.			86400	IN	DS	16378 5 1 4BAC0A7FB788380C75B12257AF422CF2F32EE828
link.			86400	IN	DS	16378 5 2 22F61A68A2F1BFE7A3B8A71DE8DB12BAC9A51D98D8D1F44C1717396E 395C180B
link.			86400	IN	DS	33631 5 1 3B75B8FA70D02DD2318B92DBAF7C19E97227CBD3
link.			86400	IN	DS	33631 5 2 98C5361E319C349C3135070138A0A8B34D5CA3CFFE87B3D27294E313 E696F101
link.			86400	IN	DS	51802 5 1 4F1558597AEC14F10D869F5F43D4905CDF6189DF
link.			86400	IN	DS	51802 5 2 2D24B7DF7EFCB55786172F5927D12F15FBB5905EC98C9FAB2525A901 5119DF29
link.			86400	IN	RRSIG	DS 8 1 86400 20210908170000 20210826160000 26838 . gPfhzPMYmQtReJMDVzBhebB8gskitaN12M1RXTfK2gAMyXmB5AsvSwNS ULfo/DneKtHxMkotgOz0TU+hNECtMe7sf8pJBVngUAKs8LfnCy+l0LRF tm45W0cAil5MghLgAa7j+ovzT9qPNJpMzqnMF6lLHXuZtB7B5nR6oJ7U +NZQmxGq950vOODtsJXFIxdkvWWvRFmnOSh1oLVgrFBFRxddZ5M3xfxZ lEYibuZPdc0zHT/1niUsB9Uq73vHuG/b1+QFMJmEQgOODpXGZETD4GLw uyHhRX5TPaKyAf5gD9CdZMRWq45ZF6k1qi7ijx1ZbHNXugNdxZbzzNOu VVcvKg==
;; Received 866 bytes from 199.9.14.201#53(b.root-servers.net) in 26 ms

netless.link.		900	IN	NS	vip4.alidns.com.
netless.link.		900	IN	NS	vip3.alidns.com.
netless.link.		86400	IN	NSEC	netlflix.link. NS RRSIG NSEC
netless.link.		86400	IN	RRSIG	NSEC 5 2 86400 20210925223649 20210826223649 38268 link. lBpvbSZiLMuYzD8cklCsCdW8hDvhYilkkll8sxW9a5rzesAM/9WrNUch kr+pDGA3DrGh/Q9fdov4+fcLT1c6uiKKYtBCIGgpTxW4pJPLpkA1bHVq jRM79hoFCZ76f94MFEYa/4iDEKb26GKQ0ITTnN/RQCNsjOjF7tRt2T/C jwQ=
;; Received 292 bytes from 64.96.2.1#53(ns2.uniregistry.info) in 20 ms

api.netless.link.	600	IN	CNAME	api-gb-lon.netless.link.
;; Received 82 bytes from 14.1.112.21#53(vip3.alidns.com) in 343 ms

**Reply by leavesster** [reply]
It' seems you get correct DNS response. The `http://shunt-api.netless/vxxxxx` should't be that slow.
could  you try `ping api.netless.link` ?

**Comment by BlackHole1** [comment, marked as answer]
After our investigation, there is a higher probability that this is due to a network fluctuation in the service provided by AliCloud last night.

Because we checked the reports from our whiteboard service last night and everything was fine

The problem you feedback, you can basically determine is the Ali cloud service problems caused by

Finally, thank you very much for your feedback, which will make us do better!
